python/example_code/bedrock/scenarios/bedrock_studio_bootstrapper.py

```diff
@@ -321,36 +321,63 @@ def _create_role(self, role_name, trust_policy, role_policy):
             logger.warning(f"Role with name '{role_name}' already exists.")
 
         logger.info(f"Attaching inline policy to '{role_name}'...")
-        try:
-            self._iam_client.put_role_policy(
-                RoleName=role_name,
-                PolicyName="InlinePolicy",
-                PolicyDocument=role_policy,
-            )
-            logger.info(f"Inline policy successfully attached.")
-        except self._iam_client.exceptions.EntityAlreadyExistsException:
-            logger.warning("Inline policy already exists.")
+        self._iam_client.put_role_policy(
+            RoleName=role_name,
+            PolicyName="InlinePolicy",
+            PolicyDocument=role_policy,
+        )
+        logger.info(f"Inline policy successfully attached.")
 
     def _create_permission_boundary(self):
+        policy_name = self._permission_boundary_policy_name
+        policy_arn = f"arn:aws:iam::{self._account_id}:policy/{self._permission_boundary_policy_name}"
+
         logger.info("=" * 80)
         logger.info("Step 3: Create Permission Boundary.")
         logger.info("-" * 80)
 
-        logger.info(
-            f"Creating permission boundary: '{self._permission_boundary_policy_name}'..."
-        )
-
+        logger.info(f"Creating permission boundary: '{policy_name}'...")
         try:
             self._iam_client.create_policy(
-                PolicyName=self._permission_boundary_policy_name,
+                PolicyName=policy_name,
                 PolicyDocument=self._get_permission_boundary(),
             )
-            logger.info(f"Permission boundary policy created.")
         except self._iam_client.exceptions.EntityAlreadyExistsException:
-            logger.warning(
-                f"Policy with name '{self._permission_boundary_policy_name}' already exists."
+            logger.warning(f"Policy with name '{policy_name}' already exists.")
+
+            policy_versions = self._iam_client.list_policy_versions(
+                PolicyArn=policy_arn
+            )["Versions"]
+            if len(policy_versions) >= 5:
+                logger.warning(
+                    f"Cannot create more than 5 versions of '{policy_name}' policy."
+                )
+                sorted_policy_versions = sorted(
+                    policy_versions, key=lambda x: x["CreateDate"]
+                )
+                oldest_non_default_version_id = next(
+                    filter(lambda x: not x["IsDefaultVersion"], sorted_policy_versions)
+                )["VersionId"]
+                logger.info(
+                    f"Deleting oldest non-default version '{oldest_non_default_version_id}'"
+                    f" of '{policy_name}' policy..."
+                )
+                self._iam_client.delete_policy_version(
```

Comment:
Is there a prompt here for the user to confirm this? Deleting versions of a policy that we may not have created seems dangerous.

Reply:
Both possible deletions are now guarded by a confirmation prompt.